fdbclient/ISingleThreadTransaction.cpp

```diff
@@ -44,12 +42,15 @@ ISingleThreadTransaction* ISingleThreadTransaction::allocateOnForeignThread(Type
 void ISingleThreadTransaction::create(ISingleThreadTransaction* tr, Type type, Database db) {
 	switch (type) {
 	case Type::RYW:
+		dynamic_cast<ReadYourWritesTransaction*>(tr)->~ReadYourWritesTransaction();
```

Comment:
If the object is already constructed on the foreign thread, why do we need to destruct and reconstruct it here?